ndarray/ndarray-core/src/commonMain/kotlin/io/kinference/ndarray/extensions/PrimitiveConvExtensions.kt

```diff
@@ -0,0 +1,190 @@
+@file:GeneratePrimitives(DataType.FLOAT, DataType.DOUBLE)
+package io.kinference.ndarray.extensions
+
+import io.kinference.ndarray.arrays.*
+import io.kinference.ndarray.arrays.pointers.PrimitivePointer
+import io.kinference.primitives.annotations.*
+import io.kinference.primitives.types.*
+import io.kinference.ndarray.extensions.utils.*
+
+
+/***
+ * Performs convolution as described by ONNX docs.
+ */
+fun PrimitiveNDArray.conv(
+    w: PrimitiveNDArray,
+    b: PrimitiveNDArray?,
+    pads: IntArray,
+    strides: IntArray,
+    dilations: IntArray,
+    groups: Int
+): PrimitiveNDArray {
+    val xShapeWithPads = getShapeWithPads(this.shape, pads)
+    val wShapeWithDilations = getShapeWithDilations(w.shape, dilations)
+
+    val resultShape = IntArray(this.shape.size) {
+        when (it) {
+            0 -> this.shape[0]
+            1 -> w.shape[0]
+            else -> (xShapeWithPads[it] - wShapeWithDilations[it] + 1) divCeil strides[it - 2]
+        }
+    }
+
+    val result = MutablePrimitiveNDArray(resultShape) { pos: IntArray -> b?.get(intArrayOf(pos[1])) ?: 0.toPrimitive() }
```

Comment:
I think that isn't very optimal way to fill array. Try to find other way for this